src/web/user_agent.rs

```diff
@@ -58,12 +58,28 @@ impl DeviceInfo {
         self.os_family == OsFamily::IOs && self.ff_version < 36
             || self.os_family != OsFamily::IOs && self.ff_version < 91
     }
+
+    /// Determine if the device is a mobile phone based on either the form factor or OS.
+    pub fn is_mobile(&self) -> bool {
+        matches!(&self.form_factor, FormFactor::Phone | FormFactor::Tablet)
+            || matches!(&self.os_family, OsFamily::Android | OsFamily::IOs)
+    }
 }
 
 /// Parse a User-Agent header into a simplified `DeviceInfo`
 pub fn get_device_info(ua: &str) -> HandlerResult<DeviceInfo> {
-    let wresult = Parser::new().parse(ua).unwrap_or_default();
-
+    let mut wresult = Parser::new().parse(ua).unwrap_or_default();
+
+    // NOTE: Firefox on iPads report back the "desktop" UA
+    // (e.g. `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/605.1.15
+    //        (KHTML, like Gecko) Version/13.1 Safari/605.1.15)`
+    // therefore we have to accept that one. This does mean that we may presume
+    // that a mac safari UA is an iPad.
```

Comment:
What does that mean for Desktop clients using Safari on macOS?

Previously Contile would return a `403 Forbidden`. See the `error_invalid_user_agent` integration test.

Reply:
It means that if you have a UA string that matches the default Mac Desktop User Agent, we presume that you're on an iPad and provide tiles accordingly. I've adjusted the integration tests to account for that.